core/vm/instructions.go

```diff
@@ -667,6 +667,24 @@ func opJumpi(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory
 }
 
 func opJumpdest(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
+        return nil, nil
+}
+
+func opJumpSub(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack, returns *Stack) ([]byte, error) {
+	pos := stack.pop()
+	returns.push(pos)
+	if !contract.validJumpdest(pos) {
+		return nil, errInvalidJump
+	}
+	*pc = pos.Uint64()
+
+	interpreter.intPool.put(pos)
+	return nil, nil
+}
+
+func opReturnSub(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack, returns *Stack) ([]byte, error) {
+    pos = returns.pop()
```

Comment:
potential panic here

Reply:
How to handle it?  Does pop() return a tuple with an error?

Reply:
See below.  No danger of panic.
"The trick is to put the code size as the first entry. So popping that item will cause a jump into the implicit zeros at the end of the code and stop."

Reply:
Idea is that this is just like returning from `main()`.  So it's not specified as an error, just a `STOP`.  Preloading the returns stack is just a cheap way to make it happen.